### DISCUSSION TITLE
Headers ignored when in rendered page in Iframe [X-Frame-Options: DENY]

### DISCUSSION THREAD

**Discussion by dgreen-ledger** [vercel/community#3542] [Help]
### Summary

I currently have a page that needs to be iframed let's call it https://verceldomain.com/iframe. 

Whenever I load the content via an iframe I get 
`Failed to load resource: the server responded with a status of 401 ()`

And I can see the following response headers including `X-Frame-Options:DENY` which is blocking the request

<img width="572" alt="Screenshot 2023-08-08 at 17 44 32" src="https://github.com/vercel/community/assets/112866118/04eb6acb-a33a-4d9b-bd05-96abe21032cb">

When hitting the https://verceldomain.com/iframe directly, I see the correct headers that I have set in NEXT 
 
<img width="855" alt="Screenshot 2023-08-08 at 17 46 27" src="https://github.com/vercel/community/assets/112866118/67445912-e0e8-47b4-9b17-4b38e3fed656">

I have tried updating the next.config and vercel.config as well as adding middleware and nothing seems to apply when the content is iframed. 

So it feels like vercel is intercepting somewhere and the server headers are not getting through. 

Does anyone have any ideas how to prevent this?

### Steps to Reproduce

- Create a page route or static html 
- Load the content in an iframe onto the page 
- Notice the header `X-Frame-Options:DENY` is set

**Comment by baxuu** [comment]
@dgreen-ledger got same issue, have you found a solution?

**Comment by JoaquinPiersigilli** [comment]
@dgreen-ledger @baxuu I'm having the exact same problem, were you able to work around it?

**Reply by JoaquinPiersigilli** [reply]
@amyegan Do you happen to know a way around this? I read your comments on this Issue [1664](https://github.com/orgs/vercel/discussions/1664) but not entirely sure if its related?

**Reply by dgreen-ledger** [reply]
We established this is issue was due to the password protection feature in vercel.
We ended up removing the password protection to test the functionality which seemed to resolve the issue.

**Reply by JoaquinPiersigilli** [reply]
Thanks for the answer, will give that a try after getting the needed permissions
